src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +74,34 @@
     }
   }
 
+  /**
+   * Get the historical price data for a given stock symbol.
+   *
+   * @param symbol stock symbol to get historical data for.
+   * @param range specified time range of the request.
+   * @param date specified date in the format YYYYMMDD.
+   * @return a list of historical price data for the given symbol.
+   */
+  public List<IexHistoricalPrice> getHistoricalPrice(final String symbol, final String range,
+      final String date) {
+    log.info("Retrieving historical price with symbol {}, range {}, date {}", symbol, range, date);
+    boolean isAllLetters = Pattern.matches("[a-zA-Z]+", symbol);
+    if (!symbol.isEmpty() && isAllLetters) {
```

Comment:
In the rest controller symbol is required, but because this is a public method, you probably want to do some sort of null check on symbol before calling ```isEmpty()```

Alternatively, you should always look to reuse useful dependencies. One of my favorites is Apache Commons
https://www.baeldung.com/apache-commons-collection-utils